app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/components/MenuItem.kt

```diff
@@ -0,0 +1,230 @@
+package com.realityexpander.tasky.agenda_feature.presentation.common
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.DropdownMenuItem
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Mail
+import androidx.compose.material.icons.filled.Textsms
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import com.realityexpander.tasky.core.presentation.common.modifiers.DP
+import com.realityexpander.tasky.core.presentation.theme.TaskyTheme
+
+@Composable
+fun MenuItem(
+    title: String,
+    icon: ImageVector? = null,
+    iconContentDescription: String? = "Menu item $title",
+    onClick: () -> Unit,
+    isSelected: Boolean = false,
+    isHighlighted: Boolean = false,
+    enabled: Boolean = true,
+    highLightColor: Color = MaterialTheme.colors.secondary,
+) {
+    val highlightColorNative = remember(isHighlighted) {
+        Color(highLightColor.copy(alpha = .9f).toArgb())
+    }
+
+    DropdownMenuItem(
+        enabled = enabled,
+        onClick = {
+            onClick()
+        },
+        modifier = Modifier
+            .background(MaterialTheme.colors.onSurface)
+            .drawBehind {
+                if (isHighlighted) drawRect(highlightColorNative)
+            }
```

Comment:
Any reason, you didn't use the `background` modifier for the highlight?

Reply:
No reason, i can change it to use the `background` instead.